### PR TITLE
added "favorite toggle" feature

### PR DESCRIPTION
**Pull request by benkauffman**
This feature is tested but seems to throw the "Contents of buffer are null" error which is a common error with the PokeGOAPI-Java API https://github.com/Grover-c13/PokeGOAPI-Java/issues?utf8=%E2%9C%93&q=Contents%20of%20buffer%20are%20null%20

### CODE REVIEW

File: src/me/corriekay/pokegoutil/windows/PokemonTab.java [hunk 1]
```diff
@@ -381,6 +387,58 @@ private void powerUpSelected() {
...
+                                    "Error toggling favorite for " + PokeHandler.getLocalPokeName(poke) + " , result: " + result.toString());
+                        }
+
+                    } catch (Exception e) {
+
+                        // if the error was "Contents of buffer are null"
+                        // it was likely successful
+                        // https://github.com/Grover-c13/PokeGOAPI-Java/issues?utf8=%E2%9C%93&q=Contents%20of%20buffer%20are%20null%20
+                        err.increment();
+                        System.out.println("Error toggling Pokémon(s) favorite! " + e.getMessage());
```
Comment (Wolfsblvt) [inline src/me/corriekay/pokegoutil/windows/PokemonTab.java]: Can you add the Name of the pokémon to that Error message, like in all other bulk operations in one of the latest bug fixes?
Just include that `PokeHandler.getLocalPokeName(poke)`.

File: src/me/corriekay/pokegoutil/windows/PokemonTab.java [hunk 2]
```diff
@@ -73,12 +74,13 @@ public PokemonTab(PokemonGo go) {
...
         this.go = go;
         JPanel topPanel = new JPanel(new GridBagLayout());
-        JButton refreshPkmn, renameSelected, transferSelected, evolveSelected, powerUpSelected;
+        JButton refreshPkmn, renameSelected, transferSelected, evolveSelected, powerUpSelected, toggleFavoriteSelected;
         refreshPkmn = new JButton("Refresh List");
         renameSelected = new JButton("Rename");
         transferSelected = new JButton("Transfer");
         evolveSelected = new JButton("Evolve");
         powerUpSelected = new JButton("Power Up");
+        toggleFavoriteSelected = new JButton("Toggle Favorite Selected");
```
Comment (Wolfsblvt) [inline src/me/corriekay/pokegoutil/windows/PokemonTab.java]: Can you remove the "Selected"? Just keep button names as short as possible for small resolutions and such.

**Comment by Wolfsblvt** [comment]
I've commented two lines where it would be cool if you change it.

On top of that, can you reformat the two big switch case statements at the end of the file like before?
It's bad to make that auto-formatting wars with commits *lol*
And I guess it's better style to have case statements with new line for the code that gets executed.

If you fix all three, I am fine with that. Really cool that you did that!

File: src/me/corriekay/pokegoutil/windows/PokemonTab.java (hunk 2 above)
Comment (benkauffman) [inline src/me/corriekay/pokegoutil/windows/PokemonTab.java]: No problem. I've updated. I'll take care of the formatting now

**Comment by benkauffman** [comment]
I've updated to match your requests. Let me know if you run into anything else,
cheers!

File: src/me/corriekay/pokegoutil/windows/PokemonTab.java [hunk 3]
```diff
@@ -706,6 +766,7 @@ private PokemonTableModel(PokemonGo go, List<Pokemon> pokes, PokemonTable pt) {
                 dustToLevelCol.add(i.getValue(), p.getStardustCostsForPowerup());
                 pokeballCol.add(i.getValue(), WordUtils.capitalize(p.getPokeball().toString().toLowerCase().replaceAll("item_", "").replaceAll("_", " ")));
                 caughtCol.add(i.getValue(), LocalDateTime.ofInstant(Instant.ofEpochMilli(p.getCreationTimeMs()), ZoneId.systemDefault()));
+                favCol.add(i.getValue(), p.isFavorite());
```
Comment (Wolfsblvt) [inline src/me/corriekay/pokegoutil/windows/PokemonTab.java]: Oh, nearly forgot. I don't know how that bool is displayed in the column, but wouldn't it be cooler to just display something if it is a favorite?

So do `(p.isFavorite()) ? "True" : ""` here.
That makes the column even better sortable and it's better to see what are favorites. With the new UI, we will add yellow stars :P

**Comment by Wolfsblvt** [comment]
Yeah, did another comment, seems like it's hidden with your latest commit :D

**Comment by benkauffman** [comment]
It was displaying as a boolean value (which is also output in the logs), which made sense to me. I have not preference on the formatting, so I updated it to match your request ;)

**Comment by Wolfsblvt** [comment]
Boolean would be fine, aber guess it's quicker to see if its a favorite or not if one contains something in the cell where the other does not.
Maybe it's just me, but should be fine now.

Okidoki, reviewed it and tested it locally, works like a charm. Thank you for that submission (:

**Comment by Wolfsblvt** [comment]
Forgot to mention, this fixes #58.

**Comment by benkauffman** [comment]
That's rad! Hopefully I can help develop some more features with you!
thanks for investing the time and building this cool app.

**Comment by Wolfsblvt** [comment]
Yeah, thanks for contributing too (: If you want to talk with us and speak about new features and what you can develop, visit our official Discord server. It's linked in the Readme here on GitHub.

**Pull request merged**